src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkInterfacePal.Android.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.Net.NetworkInformation
+{
+    internal static class NetworkInterfacePal
+    {
+        /// Returns objects that describe the network interfaces on the local computer.
+        public static NetworkInterface[] GetAllNetworkInterfaces() => GetAndroidNetworkInterfaces();
+
+        public static bool GetIsNetworkAvailable()
+        {
+            foreach (var ni in GetAndroidNetworkInterfaces())
+            {
+                if (ni.NetworkInterfaceType == NetworkInterfaceType.Loopback
+                    || ni.NetworkInterfaceType == NetworkInterfaceType.Tunnel)
+                {
+                    continue;
+                }
+                if (ni.OperationalStatus == OperationalStatus.Up)
+                {
+                    return true;
```

Comment:
Seems quite wasteful to allocate the array with complex types to return true, LoopbackInterfaceIndex has a similar problem. Perhaps GetAndroidNetworkInterfaces version with a predicate would be better.

Reply:
I tried implementing your suggestion. The code is a bit more complex now but there's no need to allocate 
 `AndroidNetworkInterface[]` in `LoopbackInterfaceIndex` and `GetIsNetworkAvailable()` anymore.